kernel: bump 5.15 to 5.15.125

graysky2 opened the pull request:
1. Add new symbols to generic config
2. Bump kernel
Changelog: https://lore.kernel.org/stable/2023080818-groin-gradient-a031@gregkh/
    
All patches automatically rebased.

graysky2 committed:
kernel: bump 5.15 to 5.15.124

Changelog: https://lore.kernel.org/stable/2023080341-curliness-salary-4158@gregkh/

1. Needed to make a change to to package/kernel/linux/modules/netsupport.mk
   due to upstream moving vxlan to its own directory[1].  @john-tho suggested
   using the the 6.1 xvlan FILES to circumvent.
2. All patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.124&id=77396fa9096abdbfbb87d63e73ad44d5621cf103

Signed-off-by: John Audia <therealgraysky@proton.me>

john-tho commented:
For the [`vxlan.ko is missing` error](https://github.com/openwrt/openwrt/actions/runs/5754070908/job/15598689422?pr=13219)
> ```
> ERROR: module '/__w/openwrt/openwrt/openwrt/build_dir/target-powerpc-openwrt-linux-musl_musl/linux-apm821xx_nand/linux-5.15.124/drivers/net/vxlan.ko' is missing.
> ```
Due to stable commit: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git/commit/?h=linux-5.15.y&id=77396fa9096abdbfbb87d63e73ad44d5621cf103
https://github.com/openwrt/openwrt/blob/b8be20c7e81de2894df8fa2b361c39bc723e4cb5/package/kernel/linux/modules/netsupport.mk#L97

We _should_ be able to use the 6.1 vxlan FILES. Untested…
```patch
diff --git a/package/kernel/linux/modules/netsupport.mk b/package/kernel/linux/modules/netsupport.mk
index 35e627002eac..0187a058258c 100644
--- a/package/kernel/linux/modules/netsupport.mk
+++ b/package/kernel/linux/modules/netsupport.mk
@@ -93,8 +93,7 @@ define KernelPackage/vxlan
 	+IPV6:kmod-udptunnel6
   KCONFIG:=CONFIG_VXLAN
   FILES:= \
-	$(LINUX_DIR)/drivers/net/vxlan.ko@lt5.18 \
-	$(LINUX_DIR)/drivers/net/vxlan/vxlan.ko@ge5.18
+	$(LINUX_DIR)/drivers/net/vxlan/vxlan.ko
   AUTOLOAD:=$(call AutoLoad,13,vxlan)
 endef
 
```

graysky2 commented:
Thanks @john-tho - I was looking into `package/network/config/vxlan/Makefile` but wasn't seeing how to adjust for the path change.  I incorporated your suggestion.

graysky2 committed:
kernel: bump 5.15 to 5.15.125

1. Add new symbols to generic config
2. Bump kernel
   Changelog: https://lore.kernel.org/stable/2023080818-groin-gradient-a031@gregkh/

   All patches automatically rebased.

Signed-off-by: John Audia <therealgraysky@proton.me>
